hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,120 @@
+# EVM Transaction Response Codes in Hedera
+
+When executing a transaction processed by the EVM (i.e., EVM Transaction) using `ContractCreate`, `ContractCall`, `EthereumTransaction`, and `ContractCallLocal` in Hedera, different response codes may be present for the top-level status after the transaction is finished. 
+These responses are found in the transaction receipt. This document's goal is to list the possible EVM transaction type-specific status codes for failure (i.e. different from SUCCESS).
+
+## Propagated EVM Transaction Response Codes
+These response codes can appear for `ContractCall`, `ContractCreate`, and `EthereumTransaction`.
+It is possible for `ContractCreate` and `ContractCall` to create child transactions, and if one of them fails with a special revert or halt reason, the status is propagated to the top-level status of the transaction.
+`ContractCall` creates child transactions by executing logic that results in another call to a function/contract.
+`ContractCreate` creates child transactions by having logic containing calls in the constructor.
+
+- **CONTRACT_REVERT_EXECUTED**
+  - The default error for contract execution being reverted. This also happens when the REVERT opcode is executed in the contract.
+
+- **INVALID_SOLIDITY_ADDRESS**
```

Comment:
This is an example of where the description in the response code's enum protobuf is not as good as it could be.  In this case even the response code _name_ is not as good as it could be.

I suggest finishing this doc "as is" and (getting it merged) and maybe later if there's interest coming back to it and fixing up some of these explanations with what's really going on.  (And fixing up the protobuf comments as well.)

Reply:
Yeah, I agree. Some of the other descriptions are also slightly refined for better readability, so in the future, it could be worth revisiting the descriptions.